pallets/msa/src/lib.rs

```diff
@@ -1357,6 +1359,89 @@ impl<T: Config> Pallet<T> {
 		Ok(result)
 	}
 
+	/// Determine the address of a contract using the CREATE2 semantics.
+	/// Cloned from pallet-revive; when pallet-revive is added as a dependency, this can be removed.
+	/// https://docs.rs/pallet-revive/0.5.0/src/pallet_revive/address.rs.html#212-224
+	fn create2(deployer: &H160, code: &[u8], input_data: &[u8], salt: &[u8; 32]) -> H160 {
+		let init_code_hash = {
+			let init_code: Vec<u8> = code.iter().chain(input_data).cloned().collect();
+			keccak_256(init_code.as_ref())
+		};
+		let mut bytes = [0; 85];
+		bytes[0] = 0xff;
+		bytes[1..21].copy_from_slice(deployer.as_bytes());
+		bytes[21..53].copy_from_slice(salt);
+		bytes[53..85].copy_from_slice(&init_code_hash);
+		let hash = keccak_256(&bytes);
+		H160::from_slice(&hash[12..])
+	}
+
+	/// Converts an MSA ID into a synthetic Ethereum address (raw 20-byte format) as follows:
+	/// [0..8]:   u64 (big endian)
+	/// [8..17]:  b"Generated"
+	/// [17..20]: all zeroes
+	pub fn msa_id_to_eth_address(id: MessageSourceId) -> H160 {
+		// Use a canned value for all but mainnet/testnet builds
+		let genesis_hash: [u8; 32] = if cfg!(any(
```

Comment:
To simplify this I think you could do it the other way around and checking to see if main-net or testnet and everything else uses the default genesis

Reply:
Flipped.